File: php/statePattern/MissionAbort.php
<?php  namespace php\statePattern;

class MissionAbort implements State {
		
	/**
	 * Mission control instance
	 * 
	 * @var MissionControl $missionControl
	 */
	public $missionControl;
			
	/**
	 * Set instance for mission control
	 * 
	 * @param MissionControl $missionControl
	 * @return void
	 */
	public function __construct(MissionControl $missionControl) {
		$this->missionControl = $missionControl;
	}
	
	/**
	 * Check all systems are stable and ready to go
	 * 
	 * @return void
	 */
	public function checkAllSystems() {
		echo "Countdown aborted.<br>";
	}
	
	/**
	 * set state in mission control if permitted
	 * 
	 * @return void
	 */
	public function startPowerUnits() {
		echo "Power units have been shut down.<br>";
	}
			
	/**
	 * Print status
	 * 
	 * @return void
	 */
	public function retractArms() {
		echo "Fuel drainage in at safe level.<br>";
	}
		
	/**
	 * update missionControl if permitted
	 * 
	 * @return void
	 */
	public function shutDown() {
		echo "All systems are to cease operations immediatly.<br>";
		$this->missionControl->liftOffHandOver($this->missionControl->getAbortProcedure());
	}
			
	/**
	 * Print negative statement
	 * 
	 * @return void
	 */
	public function rocketBooster() {
		echo "Launch is totally aborted.<br>";
		
	}
}

File: java/statePattern/MissionAbort.java
package statePattern;

public class MissionAbort implements State {
	private MissionControl missionControl;

	public MissionAbort(MissionControl missionControl) {
		this.missionControl = missionControl;
	}

	@Override
	public void checkAllSystems() {
		System.out.println("Systems have been cleared");
	}

	@Override
	public void startPowerUnits() {
		System.out.println("Power units have been started");
	}

	@Override
	public void retractArms() {
		System.out.println("Arms and fuel connections have been retracted.");

	}

	@Override
	public void shutDown() {
		System.out.println("All systems are to cease operations immediatly");
		this.missionControl.setState(this.missionControl.getAbortProcedure());
	}

	@Override
	public void rocketBooster() {
		System.out.println("We have lift off.");
		this.missionControl.setState(this.missionControl.getLiftOffStatus());

	}

}
